Brainfuck/ruby/main.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+module Bf
+  def self.run(prog)
+    data_stack = Hash.new { 0 }
+    data_pointer = 0
+
+    instruction_pointer = 0
+    instruction_stack = []
+
+    while instruction_pointer < prog.length
+      case prog[instruction_pointer]
+      when ">"
+        data_pointer += 1
+      when "<"
+        data_pointer -= 1
+      when "+"
+        data_stack[data_pointer] += 1
+      when "-"
+        data_stack[data_pointer] -= 1
+      when "."
+        STDOUT.putc(data_stack[data_pointer].chr)
+      when ","
+        data_stack[data_pointer] = STDIN.getc.ord
+      when "["
+        stack = 1
+        ending = instruction_pointer + 1
+
+        while stack > 0
+          stack += { "[" => 1, "]" => -1 }.fetch(prog[ending], 0)
```

Comment:
I don't understand what this is doing, could we simplify this?

Reply:
Simplified